pkg/kubecost/allocation.go

```diff
@@ -743,49 +743,64 @@ func NewAllocationSet(start, end time.Time, allocs ...*Allocation) *AllocationSe
 // succeeds, the allocation is marked as a shared resource. ShareIdle is a
 // simple flag for sharing idle resources.
 type AllocationAggregationOptions struct {
-	FilterFuncs       []AllocationMatchFunc
-	IdleByNode        bool
-	LabelConfig       *LabelConfig
-	MergeUnallocated  bool
-	SharedHourlyCosts map[string]float64
-	ShareFuncs        []AllocationMatchFunc
-	ShareIdle         string
-	ShareSplit        string
-	SplitIdle         bool
+	AllocationResourceTotalsStore ResourceTotalsStore
```

Comment:
I wrestled with this one. Putting it on the query, itself, was a simple way to "inject" the functionality into the query. Nice for me, writing the feature; bad in almost every other way, probably. So as I re-read the code, I think this belongs elsewhere. But where?

We could add it as a parameter on creation of an `ETLStore`. That, however, has the downside of actually requiring more threading of the type through constructors (a couple places per implementation of `ETLStore`, which is now 3 and growing) as opposed to living in two places (`QueryOptions` and `AggregationOptions`). Perhaps we'd come up with more ideas, given the opportunity to refactor ETL into a more sensible, hierarchical pattern altogether. But that's my thought.

Good call-out.

Reply:
I'm fairly happy with requiring these in `QueryOptions` because it makes a data dependency in the query explicit (side note for KCM, we could have a separate options struct for summary queries and just embed the regular query struct :man_shrugging:). Seeing the dependency in `AggregationOptions`, though, is what's bothering me and probably harks back to our "should aggregateby really be called aggregateby" discussion we had months ago.